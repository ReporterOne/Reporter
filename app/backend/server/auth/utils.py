from datetime import timedelta, datetime
from typing import List

import jwt
from fastapi import Depends, HTTPException, Security
from fastapi.security import OAuth2PasswordBearer, SecurityScopes
from jwt import ExpiredSignatureError, PyJWTError
from pydantic import BaseModel, ValidationError
from sqlalchemy.orm import Session
from starlette.status import HTTP_401_UNAUTHORIZED

from db.crud import get_user_by_username
from db.database import verify_password, get_db
from db.schemas import User
from .consts import SECRET_KEY, ALGORITHM



class Token(BaseModel):
    """Token model."""
    access_token: str
    token_type: str


class TokenData(BaseModel):
    """Token data model."""
    username: str = None
    user_id: int = None
    scopes: List[str] = []


oauth2_scheme = OAuth2PasswordBearer(
    tokenUrl="/api/login",
    scopes={
        "personal": "main permissions.",
    },
)


def authenticate_user(db: Session, username: str, password: str) -> User:
    """Authenticate the user and return it from the db."""
    user = get_user_by_username(db, username)
    if not user:
        return None
    if not verify_password(password, user.password):
        return None
    return user


def create_access_token(*, data: dict, expires_delta: timedelta = None) -> str:
    """Create an access token and return the encoding of it."""
    to_encode = data.copy()
    if expires_delta:
        expire = datetime.utcnow() + expires_delta
    else:
        expire = datetime.utcnow() + timedelta(minutes=15)
    to_encode.update({"exp": expire})
    encoded_jwt = jwt.encode(to_encode, SECRET_KEY, algorithm=ALGORITHM)
    return encoded_jwt


async def get_current_user(
    security_scopes: SecurityScopes,
    token: str = Depends(oauth2_scheme),
    db: Session = Depends(get_db)
) -> User:
    """Get the current user from the given token."""
    if security_scopes.scopes:
        authenticate_value = f'Bearer scope="{security_scopes.scope_str}"'
    else:
        authenticate_value = f"Bearer"

    credentials_exception = HTTPException(
        status_code=HTTP_401_UNAUTHORIZED,
        detail="Could not validate credentials",
        headers={"WWW-Authenticate": authenticate_value},
    )
    try:
        payload = jwt.decode(token, SECRET_KEY, algorithms=[ALGORITHM])
        username: str = payload.get("sub")
        if username is None:
            raise credentials_exception
        user_id: int = payload.get("id")
        if user_id is None:
            raise credentials_exception
        token_scopes = payload.get("scopes", [])
        token_data = TokenData(scopes=token_scopes, username=username,
                               user_id=user_id)
    except ExpiredSignatureError:
        raise HTTPException(
            status_code=HTTP_401_UNAUTHORIZED,
            detail="Token is expired",
            headers={"WWW-Authenticate": authenticate_value},
        )

    except (PyJWTError, ValidationError):
        raise credentials_exception

    user = get_user_by_username(db, username=token_data.username)
    if user is None:
        raise credentials_exception
    for scope in security_scopes.scopes:
        if scope not in token_data.scopes:
            raise HTTPException(
                status_code=HTTP_401_UNAUTHORIZED,
                detail="Not enough permissions",
                headers={"WWW-Authenticate": authenticate_value},
            )
    return user


async def get_current_user_secured(
    current_user: User = Security(get_current_user, scopes=["personal"])
) -> User:
    """Get the current user with secured scope."""
    return current_user
