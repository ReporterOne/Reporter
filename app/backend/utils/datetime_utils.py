from datetime import timedelta

def daterange(start_date, end_date):
    if end_date is None:
        return [start_date]

    for n in range(int ((end_date - start_date).days)):
        yield start_date + timedelta(n)