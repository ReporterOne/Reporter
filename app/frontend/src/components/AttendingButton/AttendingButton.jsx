import React, { useState, useCallback } from "react";
import { AutoSizer } from 'react-virtualized';

import styled from 'styled-components';
import Arrows from '~/assets/Arrows.svg';

import {
  SVGIcon,
  innerShaddow,
  theme,
} from '~/components/common';

import posed from 'react-pose';
import {motion, useAnimation} from "framer-motion";

const ContainerHeight = 60;
const rectangleMargin = 15;
const outlinePadding = 3;
const circleDiameter = ContainerHeight - outlinePadding * 2;


const PosedRRoundedRectangle = posed.div({
  notHere: {
    backgroundColor: theme.notApproved
  },
  here: {
    backgroundColor: theme.approved
  },
  notDecided: {
    backgroundColor: theme.main
  },
});

const PosedArrowsContainer = posed.div({
  decided: { scale: 0 },
  notDecided: { scale: 1 }
});

const Container = styled.div`
  display: flex;
  align-items:center;
  flex:1;
  flex-direction: row;
`;

const RoundedRectangle = styled(PosedRRoundedRectangle)`
  align-items: center;
  position: absolute;
  display: flex;
  flex: 1;
  border-radius: ${ContainerHeight / 2}px;
  ${innerShaddow[4]}
`;

const InnerContainer = styled.div`
  display: flex;
  align-items: center;
`;

const OuterContainer = styled.div`
  display: flex;
  position: relative;
  flex: 1;
  height: ${ContainerHeight}px;
  margin-left: ${rectangleMargin}px; 
  margin-right: ${rectangleMargin}px;
`;

const Spacer = styled.div`
  flex: 1;
`;

const Circle = styled(motion.div)`
  height: ${circleDiameter}px;
  width: ${circleDiameter}px;
  background-color: #ffffff;
  border-radius: 50%;
  display: inline-block;
  z-index: 1;
`;

const ArrowsContainer = styled(PosedArrowsContainer)`
  flex:1;
  height: ${ContainerHeight}px;
  align-items: center;
  display: flex;
  justify-content: center;
`;

const ArrowsLeft = styled(SVGIcon)`
  flex:1;
  height: 60%;
  transform: scaleX(-1);
  fill: #F15A24;

`;

const ArrowsRight = styled(SVGIcon)`
  flex:1;
  fill: #22B573;
  height: 60%;

`;
const AttendenceValue = styled.span`
  color: white;
  display: flex;
  font-weight: normal;
  font-size: 20px;
  white-space: nowrap;
  text-align: center;
`;


const attendenceStatus = {
  here: "Attending",
  notHere: "Missing",
  notDecided: ""
};
const ANIMATION_TIME = 0.5;
const DEACCELERATION = 2;

const AttendingButton = ({missingReason, onChange}) => {
  const [pose, changePose] = useState("notDecided");
  const controls = useAnimation();
  const onDragEnd = useCallback((containerWidth) => (event, info) => {
    const endPos = info.point.x + info.velocity.x * ANIMATION_TIME + ANIMATION_TIME*ANIMATION_TIME*DEACCELERATION*0.5;

    let state = "notDecided";
    if (endPos >= ((containerWidth - circleDiameter) * 0.5) + 70) {
      state = 'here';
    }
    else if (endPos <= ((containerWidth - circleDiameter) * 0.5) - 70) {
      state = 'notHere';
    }
    controls.start(state);
    changePose(state);
  });

  const handleChange = useCallback((state) => {
    changePose(state);
    setTimeout(() => {
      onChange(state);
    }, theme.handleSpeed * 1000);
  }, [changePose, onChange]);

  return (
    <Container>
      <OuterContainer>
        <AutoSizer>
          {({ height, width }) => (
              <InnerContainer style={{ height, width }}>
                <RoundedRectangle pose={pose} style={{ height, width }}>
                  <Spacer />
                  <ArrowsContainer pose={pose === "notDecided" ? "notDecided" : "decided"}>
                    <ArrowsLeft src={Arrows} />
                  </ArrowsContainer>
                  <Spacer />
                  <AttendenceValue>
                    {pose === "notHere" ? missingReason || attendenceStatus[pose] : attendenceStatus[pose]}
                  </AttendenceValue>
                  <Spacer />
                  <ArrowsContainer pose={pose === "notDecided" ? "notDecided" : "decided"}>
                    <ArrowsRight src={Arrows} />
                  </ArrowsContainer>
                  <Spacer />
                </RoundedRectangle>
                <Circle key={width}
                        drag={"x"} dragConstraints={{left: outlinePadding, right: width - circleDiameter - outlinePadding }}
                        variants={{
                          notHere: {x: outlinePadding},
                          here: {x: width - circleDiameter - outlinePadding},
                          notDecided: {x: ((width - circleDiameter) * 0.5)}
                        }}
                        initial="notDecided"
                        dragElastic={false}
                        animate={controls}
                        onDragEnd={onDragEnd(width)}
                />
              </InnerContainer>
            )
          }
        </AutoSizer>
      </OuterContainer>
    </Container>
  );
}

export default AttendingButton;