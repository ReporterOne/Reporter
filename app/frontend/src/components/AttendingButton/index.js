export { default as default } from "./AttendingButton.jsx";