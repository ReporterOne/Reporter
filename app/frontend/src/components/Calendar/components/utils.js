import format from 'date-fns/format';
export const ISO_FORMAT = 'yyyy-MM-dd';


export const formatDate = (date) => {
  return format(date, ISO_FORMAT);
};
