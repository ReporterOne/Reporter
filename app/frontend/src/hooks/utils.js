import {updateLogin} from "~/actions/general";
import AuthService, {PermissionsError} from "~/services/auth";

export const logoutIfNoPermission = async (callback, dispatch) => {
  try {
    await callback();
  } catch (e) {
    if (e instanceof PermissionsError) {
      await AuthService.logout();
      dispatch(updateLogin(false));
    } else {
      throw e;
    }
  }
};
