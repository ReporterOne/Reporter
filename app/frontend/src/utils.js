
export const users = [
  {
    name: "Ariel Domb",
    avatar: {
      kind: 12
    },
    status: "here"
  },
  {
    name: "Elran Shefer",
    avatar: {
      kind: 13
    },
    status: "not_answered"
  },
  {
    name: "Michael Tugendhaft",
    avatar: {
      kind: 14
    },
    status: "not_here",
    reason: "חולה"
  },
  {
    name: "Osher De Paz",
    avatar: {
      kind: 15
    },
    status: "not_here",
    reason: "חולה/ת נפש"
  },
  {
    name: "Nimrod Erez",
    avatar: {
      kind: 17
    },
    status: "here"
  },
  {
    name: "Ido Azulay",
    avatar: {
      kind: 18
    },
    status: "here"
  },
  {
    name: "Asaf Shavit",
    avatar: {
      kind: 19
    },
    status: "not_answered"
  },
  {
    name: "Asaf Shavit",
    avatar: {
      kind: 19
    },
    status: "not_answered"
  },
  {
    name: "Asaf Shavit",
    avatar: {
      kind: 19
    },
    status: "not_answered"
  },
]