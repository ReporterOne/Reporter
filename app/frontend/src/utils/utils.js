export const HERE = 'here';
export const NOT_ANSWERED = 'not_answered';
export const NOT_HERE = 'not_here';
export const ANSWERED = 'answered';


export const deepCopy = (a) => JSON.parse(JSON.stringify(a));

export const iteratePrevCurrentNext = (iterator, callback) => {
  const toRet = [];
  const len = iterator.length;
  for (let i = 0; i < len; i++) {
    if (i === 0) toRet.push(callback({prev: null, current: iterator[i], next: iterator[i + 1]}, i));
    else if (i === len - 1) toRet.push(callback({prev: iterator[i - 1], current: iterator[i], next: null}, i));
    else toRet.push(callback({prev: iterator[i - 1], current: iterator[i], next: iterator[i + 1]}, i));
  }
  return toRet;
};
