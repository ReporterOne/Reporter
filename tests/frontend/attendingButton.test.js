import React from 'react';
import {render, fireEvent, act} from '@testing-library/react';
import {shallow, mount} from 'enzyme';
import AttendingButton, {Circle} from '~/components/AttendingButton/AttendingButton';
import {createBubbledEvent} from './utils/events';


describe('Test Attending Button', () => {
  test('Basic tests', () => {
    const func = jest.fn();
    fetch.mockResponse(() => new Promise(() => {
      return '<circle cx="50" cy="50" r="40" stroke="black" stroke-width="3" fill="red" />';
    }));
    act(() => {
      const {container} =
        render(<AttendingButton missingReason="test" onChange={func}/>);
      expect(container).toMatchSnapshot();
    });
  });
  test('Test missing', () => {
    const func = jest.fn();
    act(() => {
      const {container} =
        render(<AttendingButton missingReason="testing" onChange={func} initialState="notHere"/>);
      expect(container).toMatchSnapshot();
    });
  });
  test('Test here', () => {
    const func = jest.fn();
    act(() => {
      const {container} =
        render(<AttendingButton missingReason="testing" onChange={func} initialState="here"/>);
      expect(container).toMatchSnapshot();
    });
  });
});
