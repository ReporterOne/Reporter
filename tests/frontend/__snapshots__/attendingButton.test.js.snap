// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Attending Button Basic tests 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-radius: 30px;
  box-shadow: inset 0 0px 28px rgba(0,0,0,0.25),inset 0 0px 10px rgba(0,0,0,0.22);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 60px;
  margin-left: 15px;
  margin-right: 15px;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c7 {
  height: 54px;
  width: 54px;
  background-color: #ffffff;
  border-radius: 50%;
  display: inline-block;
  z-index: 1;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: normal;
  font-size: 20px;
  white-space: nowrap;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        style="overflow: visible; height: 0px; width: 0px;"
      >
        <div
          class="c2"
          style="height: 0px; width: 0px;"
        >
          <div
            class="c3"
            style="height: 0px; width: 0px; background-color: rgb(71, 37, 165);"
          >
            <div
              class="c4"
            />
            <div
              class="c5"
              style="transform: none;"
            />
            <div
              class="c4"
            />
            <span
              class="c6"
            />
            <div
              class="c4"
            />
            <div
              class="c5"
              style="transform: none;"
            />
            <div
              class="c4"
            />
          </div>
          <div
            class="c7 AttendingHandle"
            style="transform: translateX(-27px) translateZ(0);"
          />
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Attending Button Test here 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-radius: 30px;
  box-shadow: inset 0 0px 28px rgba(0,0,0,0.25),inset 0 0px 10px rgba(0,0,0,0.22);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 60px;
  margin-left: 15px;
  margin-right: 15px;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c7 {
  height: 54px;
  width: 54px;
  background-color: #ffffff;
  border-radius: 50%;
  display: inline-block;
  z-index: 1;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: normal;
  font-size: 20px;
  white-space: nowrap;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        style="overflow: visible; height: 0px; width: 0px;"
      >
        <div
          class="c2"
          style="height: 0px; width: 0px;"
        >
          <div
            class="c3"
            style="height: 0px; width: 0px; background-color: rgb(34, 181, 115);"
          >
            <div
              class="c4"
            />
            <div
              class="c5"
              style="transform: scale(0) translateZ(0);"
            />
            <div
              class="c4"
            />
            <span
              class="c6"
            >
              Attending
            </span>
            <div
              class="c4"
            />
            <div
              class="c5"
              style="transform: scale(0) translateZ(0);"
            />
            <div
              class="c4"
            />
          </div>
          <div
            class="c7 AttendingHandle"
            style="transform: translateX(-57px) translateZ(0);"
          />
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`Test Attending Button Test missing 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
}

.c3 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  border-radius: 30px;
  box-shadow: inset 0 0px 28px rgba(0,0,0,0.25),inset 0 0px 10px rgba(0,0,0,0.22);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 60px;
  margin-left: 15px;
  margin-right: 15px;
}

.c4 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
}

.c7 {
  height: 54px;
  width: 54px;
  background-color: #ffffff;
  border-radius: 50%;
  display: inline-block;
  z-index: 1;
}

.c5 {
  -webkit-flex: 1;
  -ms-flex: 1;
  flex: 1;
  height: 60px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c6 {
  color: white;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-weight: normal;
  font-size: 20px;
  white-space: nowrap;
  text-align: center;
}

<div>
  <div
    class="c0"
  >
    <div
      class="c1"
    >
      <div
        style="overflow: visible; height: 0px; width: 0px;"
      >
        <div
          class="c2"
          style="height: 0px; width: 0px;"
        >
          <div
            class="c3"
            style="height: 0px; width: 0px;"
          >
            <div
              class="c4"
            />
            <div
              class="c5"
              style="transform: scale(0) translateZ(0);"
            />
            <div
              class="c4"
            />
            <span
              class="c6"
            />
            <div
              class="c4"
            />
            <div
              class="c5"
              style="transform: scale(0) translateZ(0);"
            />
            <div
              class="c4"
            />
          </div>
          <div
            class="c7 AttendingHandle"
          />
        </div>
      </div>
      <div
        class="resize-triggers"
      >
        <div
          class="expand-trigger"
        >
          <div
            style="width: 1px; height: 1px;"
          />
        </div>
        <div
          class="contract-trigger"
        />
      </div>
    </div>
  </div>
</div>
`;
